modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/EntityModelSchemaBasedEdmProvider.java

```diff
@@ -325,6 +339,29 @@ private FullQualifiedName _getFullQualifiedName(EntityModel entityModel) {
 		return new FullQualifiedName(_NAMESPACE + "." + entityModel.getName());
 	}
 
+	private void _handleComplexType(
+		Map<String, ObjectValuePair<Integer, CsdlComplexType>>
+			complexTypeNameDepth,
+		CsdlComplexType csdlComplexType,
+		ComplexEntityField curComplexEntityField, int depth) {
+
+		if (complexTypeNameDepth.containsKey(csdlComplexType.getName())) {
```

Comment:
I think you could use the `compute` method. What do you think about:

```
complexTypeNameDepth.compute(
	csdlComplexType.getName(),
	(key, value) -> {
		if (value == null || value.getKey() > depth) {
			return new ObjectValuePair<>(depth, csdlComplexType1);
		}

		return value;
	});
```

Reply:
In fact, the method seems more difficult than it really is. I would recommend to replace with that instead of using that method